server/src/main/java/org/opensearch/search/GenericSearchExtBuilder.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.search;
+
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public final class GenericSearchExtBuilder extends SearchExtBuilder {
+
+    public final static ParseField EXT_BUILDER_NAME = new ParseField("generic_ext");
+
+    private final Object simpleVal;
```

Comment:
I believe could significantly reduce the implementation of this builder by operation over just two values:

```
private final Object value;
private final ValueType valueType;
```

The valueType + value would give you the right type (afaik you need typecast only in one place).

Reply:
I'll give it a try!